app/questionnaire/questionnaire_manager.py

```diff
@@ -85,14 +85,15 @@ def get_rendering_context(self, location, is_valid=True):
                 self.build_state(location, get_answers(current_user))
 
         if self.state:
-            self._plumbing_preprocessing()
+            context = self._get_schema_template_context()
+            self._plumb(context, self.state)
             self._conditional_display(self.state)
 
         # look up the preprocessor and then build the view data
         return build_questionnaire_model(self._json, self.state)
 
     def get_summary_rendering_context(self):
```

Comment:
Considering moving all of the rendering context thus plumbing related logic out into it's own file. Apart from that the `piping` has been stripped back to only what is required